### PR TITLE
Follow proxy no proxy host configuration

### LINKED ISSUE #744: The proxy usage should follow the no proxy host list rule
### Your checklist for this issue

- [x] Jenkins version : 2.204.2

- [x] Plugin version : 2.41

- [x] OS : linux

### Description

The remote call to slack server uses the proxy configured in jenkins even if slack server is in the 'no proxy hosts' list

### PR DESCRIPTION
**Pull request by lionelraeymaekers**
Fixes #744 If the slack url is defined in the no proxy host configuration in jenkins, no proxy should be used

### CODE REVIEW

**Comment by lionelraeymaekers** [comment]
I'm sorry i'm now sure how to link this PR to the issue I created #744

**Review by timja — APPROVED** [review]
Have you tested this and it works for you?

**Comment by lionelraeymaekers** [comment]
you should review the changes if you can
I don't really have the environment needed to build and test this sorry and I'm a little bit rusty on java.

**Comment by timja** [comment]
> @timja you should review the changes if you can
> I don't really have the environment needed to build and test this sorry and I'm a little bit rusty on java.

I don't have a way to test this either, and proxy configuration relies on someone who is blocked by this testing.

So please get this tested and confirmed and if so I'll merge...

File: src/main/java/jenkins/plugins/slack/HttpClient.java
Comment (jetersen) [inline src/main/java/jenkins/plugins/slack/HttpClient.java:57]: Not sure how the proxy host matcher works but surely this won't match `https://{teamDomain}.slack.com`

https://github.com/jenkinsci/slack-plugin/blob/610381174a22d9c3a4aac660bd8df01b7b9a860c/src/main/java/jenkins/plugins/slack/StandardSlackService.java#L198-L205

**Comment by timja** [comment]
tested now?

**Comment by lionelraeymaekers** [comment]
@timja Yes, I think this is a more elegant solution to the problem, it'll handle custom url defined in the plugin's configuration too.
It follows correctly the no proxy hosts configured in jenkins

**Comment by timja** [comment]
https://github.com/jenkinsci/slack-plugin/releases/tag/slack-2.45

**Pull request merged**
